gitlabform/gitlabform/core.py

```diff
@@ -289,10 +289,16 @@ def process_project_settings(self, project_and_group, configuration):
     @if_in_config_and_not_skipped
     def process_project_push_rules(self, project_and_group, configuration):
         push_rules = configuration['project_push_rules']
-        logging.debug("Project push rules settings BEFORE: %s", self.gl.get_project_push_rules(project_and_group))
-        logging.info("Setting project push rules: %s", push_rules)
-        self.gl.put_project_push_rules(project_and_group, push_rules)
-        logging.debug("Project push rules AFTER: %s", self.gl.get_project_push_rules(project_and_group))
+        pid = self._get_project_id(project_and_group)
```

Comment:
We are using `gitlabform/gitlab` internally at Egnyte as a library in some places. That's why we would prefer to keep the methods there self-contained. And because of that can you please move getting the pid inside them again, like it was done in `put_project_push_rules` before, @grzesuav ?

We plan to stop doing that but until that happens I would like to not change the pattern.